lisa/sut_orchestrator/hyperv/source.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import Any, List, Type
+
+from lisa import schema
+from lisa.node import RemoteNode
+from lisa.tools.ls import Ls
+from lisa.tools.remote_copy import RemoteCopy
+from lisa.tools.unzip import Unzip
+from lisa.util import InitializableMixin, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import LocalSourceSchema, SourceFileSchema, SourceSchema
+
+
+class Source(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(self, runbook: SourceSchema, **kwargs: Any) -> None:
+        super().__init__(runbook=runbook)
+        self._log = get_logger(
+            "source", self.__class__.__name__, parent=kwargs.get("parent_logger")
+        )
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return SourceSchema
+
+    def download(self, server: RemoteNode) -> List[PurePath]:
+        raise NotImplementedError()
+
+
+class LocalSource(Source):
+    def __init__(self, runbook: LocalSourceSchema, **kwargs: Any) -> None:
```

Comment:
the `__initi__` method do nothing now, it can be deleted totally.

Reply:
done.